src/HouseScene.cpp

```diff
@@ -146,28 +145,53 @@ void HouseScene::Draw(sf::RenderTarget& target) {
     // Draw Room and Grid
     house_render_texture.setView(house_view);
     house_render_texture.clear();
-    map.Draw(house_render_texture, tile_map);
+
+    for(const auto &tile_layer : map.GetTileLayers()) {
+        for(const auto &tile : tile_layer.tiles) {
+            sf::Sprite sprite_to_draw(tile_map.GetSprites()[tile.tile_map_index]);
+            sprite_to_draw.setRotation(tile.rotation);
+            int half_tile_size = tile_map.GetSpriteSize() / 2;
```

Comment:
Can hoist this out of both these for loops, the lambda and a later for loop

Reply:
Not exactly sure what you mean. I was thinking of creating a new iterator by joining all of the tile lists together. that way there would be one single lambda here. It looks like boost might handle this however I'm not sure if there is a better way to express this. Surely we would not have to bring boost in just to handle a raw loop in a raw loop... would we?

Reply:
Ah yes, way too terse :) Was just pointing out that `half_tile_size` is defined a few times, very much  a micro-optimisation.

> Surely we would not have to bring boost in just to handle a raw loop in a raw loop... would we?

I'm not entirely sure which Boost bit you're referring to, but I imagine that if you wanted to keep the iteration at this level it would either have this nested iteration internally, or if you wanted to change the in-memory to something resembling [ECS](https://en.wikipedia.org/wiki/Entity_component_system) and have the `Tiles` owned by a single vector somewhere, with only references from the `TileLayer` to that.

The other option to consider is that `TileLayer` could implement `Draw`, with the `house_render_texture` as the `RenderTarget &`, so that loop would be single depth at each individual level in the hierarchy.